packages/devcmd/src/process/ProcessExecutor.ts

```diff
@@ -165,21 +260,52 @@ export class ProcessExecutor {
       },
     });
 
-    const code = await childProcessCompletion(childProcess);
-
-    if (code !== 0) {
-      const commandName = formatProcessCommand(processInfo, identity, bold);
-      const message = `Process ${commandName} exited with status code ${code}`;
-      this.printError(message);
-      throw new Error(message);
+    const { exitCode, exitSignal } = await childProcessCompletion(childProcess);
+
+    if (exitCode !== 0) {
+      const message = formatNonZeroExitCodeMessage(processInfo, exitCode);
+
+      switch (nonZeroExitCodeHandling) {
```

Comment:
Now we have this kind of `switch` three times - maybe it's time to extract a function.

Reply:
Ah good catch! I had been thinking about doing some more cleanup to pill out more commonalities.
I'm not sure whether the switch statement can be reasonably extracted and make things shorter and/or easier to understand, but I'll check it out 👍

Reply:
Alright, I deduplicated some stuff, would be great if you could give it another look @schierla :)

Reply:
Yes, this looks similar to what I had in mind. We might have passed an optional parameter `additionalErrorOutput: string` to `handleExitInfo` instead of the `nonZeroExitCodeMessageCreator: () => string`, but solution is also fine.